lib/postProcess.js

```diff
@@ -47,9 +47,14 @@ postProcess._fetchRelatedResources = function(request, mainResource, callback) {
 
   resourcesToFetch = Object.keys(resourcesToFetch).map(function(type) {
     var ids = resourcesToFetch[type];
-    var urlJoiner = "&filter[id]=";
-    ids = urlJoiner + ids.join(urlJoiner);
-    return jsonApi._apiConfig.pathPrefix + type + "/?" + ids;
+    var qsField = "filter[id]=";
+    var urlJoiner = "&" + qsField;
+    ids = qsField + ids.join(urlJoiner);
+    var uri = jsonApi._apiConfig.pathPrefix + type + "/?" + ids;
+    if (request.route.query) {
+      uri += "&" + request.route.query;
+    }
+    return uri;
```

Comment:
I've changed it back to how it was before but we still need to pass the query string to the related resources. I am not sure how the test was working before... :confounded: